ibis/tests/expr/test_signature.py

```diff
@@ -4,71 +4,50 @@
 import pytest
 from toolz import identity
 
-import ibis.expr.rules as rlz
-from ibis.common.exceptions import IbisTypeError
-from ibis.expr.signature import (
-    Annotable,
-    Argument,
-    Optional,
-    Parameter,
-    Validator,
-)
+from ibis.expr.signature import Annotable, Optional, Parameter, Validator
+
 
+class ValidatorFunction(Validator):
+    def __init__(self, fn):
+        self.fn = fn
 
-class IsInt(Validator):
     def __call__(self, arg, **kwargs):
-        if not isinstance(arg, int):
-            raise TypeError(int)
-        return arg
+        return self.fn(arg, **kwargs)
 
 
-class IsFloat(Validator):
+class InstanceOf(Validator):
+    def __init__(self, typ):
+        self.typ = typ
+
     def __call__(self, arg, **kwargs):
-        if not isinstance(arg, float):
-            raise TypeError(float)
+        if not isinstance(arg, self.typ):
+            raise TypeError(self.typ)
         return arg
 
 
+IsAny = InstanceOf(object)
+IsBoolean = InstanceOf(bool)
+IsFloat = InstanceOf(float)
+IsInteger = InstanceOf(int)
```

Comment:
Let's calls this `IsInt` to match the `int`, same with `IsBool`/`bool` and `IsStr`/`str`